frontend/server/src/main/java/org/pytorch/serve/wlm/WorkerThread.java

```diff
@@ -174,7 +175,11 @@ public void run() {
             logger.error("Out of memory error when creating workers", oom);
             status = HttpResponseStatus.INSUFFICIENT_STORAGE;
         } catch (Throwable t) {
-            logger.warn("Backend worker thread exception.", t);
+            if (state == WorkerState.WORKER_SCALED_DOWN || state == WorkerState.WORKER_STOPPED) {
```

Comment:
you mentioned this in the commit text but can you explain a bit more? I just want to make sure there are no race conditions hidden here

Reply:
This change is not required anymore after the lock related handling done in Model.java. Removed this condition.